chainercv/experimental/links/model/fcis/fcis_resnet101.py

```diff
@@ -0,0 +1,234 @@
+        if self.iter2:
+            # 2nd Iteration
+            # get rois2 for more precise prediction
+            roi_ag_locs = roi_ag_locs.array
+            mean = self.xp.array(self.loc_normalize_mean)
+            std = self.xp.array(self.loc_normalize_std)
+            roi_loc = roi_ag_locs[:, 1, :]
+            roi_loc = (roi_loc * std + mean).astype(np.float32)
+            rois2 = loc2bbox(rois, roi_loc)
+
+            rois2[:, 0::2] = self.xp.clip(rois2[:, 0::2], 0, img_size[0])
+            rois2[:, 1::2] = self.xp.clip(rois2[:, 1::2], 0, img_size[1])
+
+            # PSROI pooling and regression
+            roi_seg_scores2, roi_ag_locs2, roi_cls_scores2 = self._pool(
+                h_cls_seg, h_ag_locs, rois2, roi_indices)
+
+            # concat 1st and 2nd iteration results
+            rois = self.xp.concatenate((rois, rois2))
+            roi_indices = self.xp.concatenate((roi_indices, roi_indices))
+            roi_seg_scores = F.concat(
+                (roi_seg_scores, roi_seg_scores2), axis=0)
+            roi_ag_locs = F.concat(
+                (roi_ag_locs, roi_ag_locs2), axis=0)
+            roi_cls_scores = F.concat(
+                (roi_cls_scores, roi_cls_scores2), axis=0)
+        return roi_seg_scores, roi_ag_locs, roi_cls_scores, rois, roi_indices
+
+    def _pool(
+            self, h_cls_seg, h_ag_locs, rois, roi_indices):
+        # PSROI Pooling
+        # shape: (n_rois, n_class*2, roi_size, roi_size)
+        pool_cls_seg = psroi_pooling_2d(
+            h_cls_seg, rois, roi_indices,
+            self.n_class * 2, self.roi_size, self.roi_size,
+            self.spatial_scale, self.group_size)
+        # shape: (n_rois, n_class, 2, roi_size, roi_size)
+        pool_cls_seg = pool_cls_seg.reshape(
+            (-1, self.n_class, 2, self.roi_size, self.roi_size))
+        # shape: (n_rois, 2*4, roi_size, roi_size)
+        pool_ag_locs = psroi_pooling_2d(
+            h_ag_locs, rois, roi_indices,
+            2 * 4, self.roi_size, self.roi_size,
+            self.spatial_scale, self.group_size)
+
+        # Classfication
+        # Group Max
+        # shape: (n_rois, n_class, roi_size, roi_size)
+        h_cls = pool_cls_seg.transpose((0, 1, 3, 4, 2))
+        h_cls = F.max(h_cls, axis=4)
+
+        # Global pooling (vote)
+        # shape: (n_rois, n_class)
+        roi_cls_scores = _global_average_pooling_2d(h_cls)
+
+        # Bbox Regression
+        # shape: (n_rois, 2*4)
+        roi_ag_locs = _global_average_pooling_2d(pool_ag_locs)
+        n_roi = roi_ag_locs.shape[0]
+        roi_ag_locs = roi_ag_locs.reshape((n_roi, 2, 4))
+
+        # Mask Regression
+        # shape: (n_rois, n_class, 2, roi_size, roi_size)
+        # Group Pick by Score
+        max_cls_indices = roi_cls_scores.array.argmax(axis=1)
+        # shape: (n_rois, 2, roi_size, roi_size)
+        roi_seg_scores = pool_cls_seg[
+            np.arange(len(max_cls_indices)), max_cls_indices]
```

Comment:
Use `self.xp.arange`.
This is faster because `np.arange` need to transfer data from CPU to GPU.